### PR TITLE
Encrypt config.yaml

### PR DESCRIPTION
**Pull request by vivian-rook**
Bug: T348476

I believe that both the worker and web configs can be combine into one file. At which point having it encrypted should simplify deployment as the git clone will have it in place, and then the main purpose of this, having the file stored in git, is achieved.

### CODE REVIEW

**Comment by vivian-rook** [comment]
@audiodude @siddharthvp I could use some help. I updated the exception to include UnicodeError though that seems to fail, works on a minimal example on py3.7 though not through tox? If I put a bare except py37-pytest will succeed but flake will fail on the bare exception. Any thoughts on what I'm doing wrong?

**Comment by siddharthvp** [comment]
@vivian-rook Looks like the error thrown is `yaml.reader.ReaderError`, which extends `yaml.error.YAMLError` which extends `Exception`. So catching `UnicodeDecodeError, UnicodeError` doesn't solve the issue. 

Instead of updating it to catch YAMLError, I think the better solution would be to give a different name to this binary file, say `config-prod.yaml` so that it doesn't interfere with local testing and unit tests.

**Comment by audiodude** [comment]
Agree with @siddharthvp. I was already confused about the comment:

```
# Is ok if we can not load config.yaml
```

How is that okay? Won't both development and production be broken?

**Comment by vivian-rook** [comment]
> Agree with @siddharthvp. I was already confused about the comment:
> 
> ```
> # Is ok if we can not load config.yaml
> ```
> 
> How is that okay? Won't both development and production be broken?

In the test code it tends towards the `default_config.yaml` file, then the code attempts to update to the config.yaml file (which in prod should be there, but in dev it is usually not). Which in looking at it feels like a very awkward way of setting it up, that the code itself tries to select if it is dev or prod? Seems like something that should be made as a startup option for which file to use. Though that would be a cleanup task for another patch (Indeed, it is yet another thing that can go away if we have it in k8s)

**Comment by audiodude** [comment]
Right, I see that now, after expanding more lines of the file. It uses `default-config.yml` and then overlays `config.yml` if it is present.

What is the plan for deploying the git-crypt key file to the production machines? Just drop it in and do `git-crypt unlock`?

**Comment by vivian-rook** [comment]
Looks like the checks are passing. @SD0001 @Audiodude the decryption key is on the bastion in /opt/

**Comment by vivian-rook** [comment]
> What is the plan for deploying the git-crypt key file to the production machines? Just drop it in and do `git-crypt unlock`?

Yeah, that's what I usually do. Usually on the bastion system for a given project. Also helps with state files for things like terraform to just have it in one place. Though object storage is nicer if we can set it up. I wonder if object storage would be sufficient for the git-crypt key as well. That has always been an "Ask a maintainer" kind of solution. Perhaps it doesn't have to be. Though that is probably a solution in search of a problem.

**Comment by siddharthvp** [comment]
> Don't you have to tell the code to read from `config-prod.yml` instead of `config.yml` now?

`config-prod.yaml` is just the name for the encrypted file. As I understand it, in production the file will be decrypted and named `config.yaml`.

File: .gitignore [hunk 1]
```diff
@@ -8,7 +8,6 @@ __pycache__/
 .venv/
 venv/
 node_modules/
-config.yaml
```
Comment (siddharthvp) [inline .gitignore:11]: I think this was in gitignore so that developers can set overrides for local testing. So this doesn't need to be removed.

Comment (vivian-rook) [reply .gitignore]: Agreed

**Comment by audiodude** [comment]
> > Don't you have to tell the code to read from `config-prod.yml` instead of `config.yml` now?
> 
> `config-prod.yaml` is just the name for the encrypted file. As I understand it, in production the file will be decrypted and named `config.yaml`.

Ok. I haven't worked with git-crypt at all, but just reading a few articles it sounds to me that it most commonly encrypts files "in place" such that they are decrypted when they are put in the filesystem by git.

**Review by audiodude — APPROVED** [review]

**Pull request merged**